### PR TITLE
implement digest

### PR DESCRIPTION
**Pull request by Aceralon**
* local digest is working.
* remote digest needs to upgrade libcalico-go, once yavirtd upgraded will works(don't need to change core).

### CODE REVIEW

File: engine/virt/image.go [hunk 1]
```diff
@@ -129,13 +129,34 @@ func (v *Virt) ImageBuildCachePrune(ctx context.Context, all bool) (reclaimed ui
...
+	if err != nil {
+		return "", err
+	}
+
+	digests, err := v.client.DigestImage(ctx, imgName, true)
+	switch {
+	case err != nil:
+		return "", err
+	case len(digests) < 1:
+		return "", nil
```
Comment (CMGS) [inline engine/virt/image.go]: 这个有二义性吧，我觉得你定义个 error 返回会更好？

Comment (Aceralon) [reply engine/virt/image.go]: done

File: engine/virt/image.go [hunk 2]
```diff
@@ -129,13 +130,34 @@ func (v *Virt) ImageBuildCachePrune(ctx context.Context, all bool) (reclaimed ui
...
+	if err != nil {
+		return "", err
+	}
+
+	digests, err := v.client.DigestImage(ctx, imgName, true)
+	switch {
+	case err != nil:
+		return "", err
+	case len(digests) < 1:
+		return "", errors.New("got no digest")
```
Comment (CMGS) [inline engine/virt/image.go]: - -不是，这不单独有个 errors.go 么，去里面定义一个不要硬编码

Comment (Aceralon) [reply engine/virt/image.go]: ok.. 刚才没找到 errors.go..

**Pull request merged**